components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/meta/MongoDBMetaExtension.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.mongodb.meta;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.model.Filters;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.metadata.AbstractMetaDataExtension;
+import org.apache.camel.component.extension.metadata.MetaDataBuilder;
+import org.apache.camel.component.mongodb.MongoDbComponent;
+import org.apache.camel.component.mongodb.conf.MongoConfiguration;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.camel.util.CastUtils.cast;
+
+public class MongoDBMetaExtension extends AbstractMetaDataExtension {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDBMetaExtension.class);
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    public MongoDBMetaExtension() {
+        this(null);
+    }
+
+    public MongoDBMetaExtension(CamelContext context) {
+        super(context);
+    }
+
+    @Override
+    public Optional<MetaData> meta(Map<String, Object> parameters) {
+        Map<String, String> textParameters = cast(parameters);
+        LOGGER.debug("Fetching mongodb meta information with params: {}", textParameters);
+
+        MongoConfiguration mongoConf = new MongoConfiguration(textParameters);
+        MongoClientURI connectionURI = new MongoClientURI(mongoConf.getMongoClientURI());
+
+        JsonNode collectionInfoRoot = null;
```

Comment:
nit: I don't think we need to initialize it to null 
```suggestion
        JsonNode collectionInfoRoot;
```